tools/calyx-writer/lib.rs

```diff
@@ -651,53 +667,48 @@ pub struct Component {
 
 impl Component {
     fn new<S: ToString>(name: S, is_comb: bool) -> Self {
-        Self {
+        let mut new_self = Self {
             attributes: vec![],
             is_comb,
             name: name.to_string(),
-            inputs: vec![
+            inputs: vec![],
+            outputs: vec![],
+            cells: vec![],
+            groups: vec![],
+            control: Control::empty(),
+            continuous_assignments: vec![],
+        };
+        if is_comb {
+            for input in [
                 Port::new_in_comp("go", 1)
-                    .with_attribute(Attribute::bool("go")),
+                    .with_attribute(Attribute::num("go", 1)),
                 Port::new_in_comp("clk", 1)
                     .with_attribute(Attribute::bool("clk")),
                 Port::new_in_comp("reset", 1)
                     .with_attribute(Attribute::bool("reset")),
-            ],
-            outputs: vec![Port::new_in_comp("done", 1)
-                .with_attribute(Attribute::bool("done"))],
-            cells: vec![],
-            groups: vec![],
-            control: Control::empty(),
-            continuous_assignments: vec![],
+            ] {
+                new_self.inputs.push(input);
+            }
+            for output in [Port::new_in_comp("done", 1)
+                .with_attribute(Attribute::num("done", 1))]
+            {
+                new_self.outputs.push(output);
+            }
```

Comment:
I don't see why this needs to be a `for` loop.

Reply:
In case there are other special output ports needed later?